assets/app/view/game/part/assignments.rb

```diff
@@ -34,29 +34,84 @@ def preferred_render_locations
+          stack_group_count = {}
+          stack_group_img = {}
+          @assignments.keys.each do |assignment|
+            stack_group = @game.assignment_stack_group(assignment)
+            img = @game.assignment_tokens(assignment, setting_for(:simple_logos, @game))
+
+            stack_group_count[stack_group] = (stack_group_count[stack_group] || 0) + 1 unless stack_group.nil?
+            stack_group_img[stack_group] = img unless stack_group.nil?
+            @assignments_to_show.append({ 'img' => img, 'count' => 1 }) if stack_group.nil?
+          end
+
+          stack_group_count.keys.each do |group|
+            @assignments_to_show.append({ 'img' => stack_group_img[group], 'count' => stack_group_count[group] })
+          end
+        end
+
+        def stack_count_background_props(axis, axis_value)
+          props = {
+            attrs: {
+              href: '/icons/stack_background.svg',
+              width: "#{ICON_RADIUS * 2}px",
+              height: "#{ICON_RADIUS * 2}px",
+            },
+          }
+
+          props[:attrs][axis] = axis_value + 10
+          axis2 = (POINTY_TALL_ITEM_LOCATIONS.any?(render_location) ? :x : :y)
+          props[:attrs][axis2] = -5
+          props
+        end
+
+        def stack_count_props(axis, axis_value)
+          props = {
+            attrs: {
+              'dominant-baseline': 'central',
+              fill: 'black',
+            },
+            style: {
+              fontSize: '20px',
+            },
+          }
+          props[:attrs][axis] = axis_value + 35
+          axis2 = (POINTY_TALL_ITEM_LOCATIONS.any?(render_location) ? :x : :y)
+          props[:attrs][axis2] = 8
+          props
         end
 
         def render_part
           axis = (POINTY_TALL_ITEM_LOCATIONS.any?(render_location) ? :y : :x)
           multiplyer = (POINTY_WIDER_ITEM_LOCATIONS.any?(render_location) ? 3 : 1)
 
           if @game
-            children = @assignments.keys.map.with_index do |assignment, index|
-              img = @game.assignment_tokens(assignment, setting_for(:simple_logos, @game))
+            children = []
+            @assignments_to_show.each.with_index do |assignment, index|
+              count = assignment[:count]
 
               props = {
                 attrs: {
-                  href: img,
+                  href: assignment[:img],
                   width: "#{ICON_RADIUS * 2}px",
                   height: "#{ICON_RADIUS * 2}px",
                 },
               }
-              props[:attrs][axis] = ((index - ((@assignments.size - 1) / 2)) * multiplyer * DELTA).round(2)
 
-              h(:image, props)
+              axis_value = ((index - ((@assignments_to_show.size - 1) / 2)) * multiplyer * DELTA).round(2)
+              props[:attrs][axis] = axis_value
+
+              children.append(h(:image, props))
+              if count > 1
+                props = stack_count_background_props(axis, axis_value)
+                children.append(h(:image, props))
```

Comment:
This looks like you're adding the assignment image twice if `count > 1`. Is that right?

Reply:
No, the first image is the assignment.
The second is the circle that is only appended in the case of a stacked assignment